earth2studio/data/ace.py

```diff
@@ -0,0 +1,219 @@
+    local : bool, optional
+        If True, data are loaded from a local directory instead of downloaded from HuggingFace. Defaults to False.
+    co2_fn : Callable[[Sequence[datetime]], np.ndarray] | None, optional
+        Optional function returning CO2 concentration (ppm) for a given UTC datetime as a numpy array. Defaults to None.
+        If provided, the global mean CO2 concentration from the source is ignored, and is computed using this function.
+        The function must accept a list of datetimes as input, and return a numpy array of CO2 concentrations of the same length.
+
+    References
+    ----------
+    - ACE2-ERA5 paper: https://arxiv.org/html/2411.11268v1
+    """
+
+    _IC_ALLOWED_YEARS = [1940, 1950, 1979, 2001, 2020]
+
+    def __init__(
+        self,
+        base_path: str = "allenai/ACE2-ERA5",
+        mode: Literal["forcing", "initial_conditions"] = "forcing",
+        forcing_subdir: str = "forcing_data",
+        ic_subdir: str = "initial_conditions",
+        local: bool = False,
+        co2_fn: Callable[[Sequence[datetime]], np.ndarray] | None = None,
+    ):
+        if mode not in ["forcing", "initial_conditions"]:
+            raise ValueError("mode must be either 'forcing' or 'initial_conditions'")
+        self._base_path = base_path
+        self._mode = mode
+        self._forcing_subdir = forcing_subdir
+        self._ic_subdir = ic_subdir
+        self._local = local
+        self.lat = ACE_GRID_LAT
+        self.lon = ACE_GRID_LON
+        self._co2_fn = co2_fn
+
+    def _validate_ic_times(self, time_list: list[datetime]) -> None:
+        for t in time_list:
+            if t.year not in self._IC_ALLOWED_YEARS:
+                raise ValueError(
+                    f"Initial condition time year {t.year} is not supported. Allowed years: {self._IC_ALLOWED_YEARS}"
+                )
+            if not (
+                t.day == 1
+                and t.hour == 0
+                and t.minute == 0
+                and t.second == 0
+                and t.microsecond == 0
+            ):
+                raise ValueError(
+                    "Initial condition times must be the first of each month at 00:00 UTC"
+                )
+
+    def __call__(
+        self,
+        time: datetime | list[datetime] | TimeArray,
+        variable: str | list[str] | VariableArray,
+    ) -> xr.DataArray:
+        time_list, var_list_e2s = prep_data_inputs(time, variable)
+
+        if self._mode == "initial_conditions":
+            self._validate_ic_times(time_list)
+
+        # Map requested Earth2Studio variable names to FME names present in files
+        try:
+            var_list_fme = [ACELexicon[v][0] for v in var_list_e2s]
+        except KeyError as e:
+            raise KeyError(f"Unknown ACE2ERA5 variable id: {e}") from e
+
+        # Determine years to fetch
+        years = sorted({t.year for t in time_list})
+        paths: list[str] = []
+        for y in years:
```

Comment:
Tqdm this with the verbose option for consistency with other data sources

https://github.com/NVIDIA/earth2studio/blob/afa4105aa79f725e5eafe1a6cf8a0bcf7b2bd5c7/earth2studio/data/arco.py#L238